lib/database/achievement.php

```diff
@@ -1099,41 +1099,32 @@ function recalculateTrueRatio($gameID)
     $query = "SELECT ach.ID, ach.Points, COUNT(*) AS NumAchieved
               FROM Achievements AS ach
               LEFT JOIN Awarded AS aw ON aw.AchievementID = ach.ID
-              WHERE ach.GameID = $gameID AND ach.Flags = 3 AND aw.HardcoreMode = 0
+              WHERE ach.GameID = $gameID AND ach.Flags = 3 AND aw.HardcoreMode = 1
               GROUP BY ach.ID";
```

Comment:
This query doesn't account for Untracked users, so the NumAchieved will sometimes be slightly higher than expected. In my tests, it didn't seem to affect the resulting calculation by more than a fraction of a point, so it's probably okay to ignore it.

Reply:
Good callout, it's a minor effect like you said but I'd rather have it be as accurate as possible so I added that to the TrueRatio calc query. I'm sure there could be an edge case on an easy achievement where the retro points could be lower than the actual depending on untracked users.